examples/echo_server.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS -XOverloadedStrings #-}
+
+import qualified Network.Freddy as Freddy
+import Data.ByteString.Lazy.Char8 (ByteString)
+
+processMessage :: Freddy.Request -> IO ()
+processMessage (Freddy.Request body replyWith failWith) = replyWith body
+
+main = do
+  fConnection <- Freddy.connect "127.0.0.1" "/" "guest" "guest"
```

Comment:
Here you want to `get` the connection once and then continue using it,
there's no need for every consumer to unwrap it every time.